src/codemagic/models/xctests/xcresulttool.py

```diff
@@ -26,6 +31,45 @@ def __init__(self, message: str, stderr: str):
 
 
 class XcResultTool(RunningCliAppMixin, StringConverterMixin):
+    @classmethod
+    @lru_cache(1)
+    def get_tool_version(cls) -> Optional[Version]:
+        # Cache the return value as it is not expected that chosen Xcode version,
+        # and consequently xcresulttool version, is changed during the execution
+        # of the action which requires xcresulttool. As there are possibly quite
+        # a few xcresulttool invocations for which we also need to know the used
+        # version, it is better to do the version check only once.
```

Comment:
```suggestion
        # Cache xcresulttool version to avoid repeated checks.
        # Assumes Xcode (and thus xcresulttool) version remains constant during execution.
```

Reply:
> Long comments are often a code smell, indicating that the code itself may be too complex or unclear.

Which is not the case here, so a short version is suffice.